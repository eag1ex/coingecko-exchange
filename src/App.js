import React from "react"
import { BrowserRouter, Switch, Route, Redirect } from "react-router-dom"
import { theme } from "@scss/material.theme"
import { ThemeProvider } from '@material-ui/core/styles'
import { MobxStore } from './store'
import Navbar from "./components/Navbar"
import Exchange from "./pages/Exchange"
import ProductDetail from "./pages/ProductDetail"
import Message from './components/Messages'
import { loggerSetting, log, delay } from 'x-utils-es'

if (process.env.NODE_ENV === 'production') {
    loggerSetting('log', 'off')
    loggerSetting('debug', 'off')
}

if (process.env.NODE_ENV === 'development') {
    log('IN_DEVELOPMENT_MODE')
}

const mobxstore = new MobxStore()
function App() {

    const [routeName, setRouteName] = React.useState({})
    const onRouteChange = (first, second) => {
        setRouteName({ first, second })
    }
 
    return (
        <BrowserRouter>

            <ThemeProvider theme={theme}>
                <Navbar routeName={routeName}/>
                <div className="container-fluid mt-3">

                    <Switch>
                    
                        <Route exact path="/" render={(props) => {
                            return (<Redirect to="/app/exchanges/paged/1"/>)                          
                        }}/> 

                        <Route exact path="/exchanges" render={(props) => {
                            return (<Redirect to="/app/exchanges/paged/1"/>)                          
                        }}/> 

                        <Route exact path="/app/exchanges/">
                            <Redirect to="/app/exchanges/paged/1"/>
                        </Route>

                        <Route exact path="/app/exchanges/paged/">
                            <Redirect to="/app/exchanges/paged/1"/>
                        </Route>

                        <Route exact path="/app/exchanges/product/">
                            <Redirect to="/app/exchanges/paged/1"/>
                        </Route>

                        <Route exact path="/app" render={(props) => {
                            return (<Redirect to="/app/exchanges/paged/1"/>)                          
                        }}/> 

                        <Route exact path="/app/:name/paged/:page">
                            <Exchange mobxstore={mobxstore} onRouteChange={(first) => onRouteChange(first)} />
                        </Route>  
 
                        <Route exact path="/app/:name/product/:productID">
                            <ProductDetail mobxstore={mobxstore} onRouteChange={(name, second) => onRouteChange('exchanges', second)} />
                        </Route>
          
                        <Route exact path="/error" render={(props) => {          
                            return (<Message type='error' value='Ups something went wrong' />)
                        }}/>                
                           
                        <Route exact path="/product/error" render={(props) => {          
                            return (<Message type='error' value='Product not found' />)
                        }}/>                
                    </Switch>    

                </div>

            </ThemeProvider>
        </BrowserRouter>
    )
}

export default App
