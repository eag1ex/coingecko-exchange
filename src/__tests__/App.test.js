/* eslint-disable no-undef */
import { loggerSetting } from 'x-utils-es'
// disable any loggind in testing
loggerSetting('log', 'off')
loggerSetting('debug', 'off')

import React from 'react'
import { render, cleanup } from '@testing-library/react'
import App from '../App'

describe('Take a snapshot', () => {
    afterEach(cleanup)
    it('<App/> should take a snapshot', () => {
        const { asFragment } = render(<App />)
    
        expect(asFragment(<App />)).toMatchSnapshot()
    })
})
