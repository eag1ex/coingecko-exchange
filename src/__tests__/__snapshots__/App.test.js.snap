// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot <App/> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1 navbar"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        />
      </div>
    </header>
  </div>
  <div
    class="container-fluid mt-3"
  >
    <div
      class="mt-5 row"
    >
      <div
        class="col-8 m-auto"
      >
        <div
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          />
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
